### PR TITLE
[WIP] Update V1 migration file, create relationships between User & Routine, Routine & Product (needs possible fix?), Routine and Product files

### PR DESCRIPTION
**Pull request by egg-bread**
Also changed jwt body to be user email, please confirm this is ok in Authorization and Authentication file!! 😗 
Still gotta finish the controllers & service files for routine and product and create request/response models for those controllers.

### CODE REVIEW

**Comment by JckXia** [comment]
Looks good! We can merge it

**Comment by egg-bread** [comment]
> Looks good! We can merge it

Still need to complete Routine related files!
In the last commit, I created the endpoints necessary for Products so far. I tested them locally but you should test them out too 😗

**Comment by egg-bread** [comment]
Routine & Product using bidirectional: https://howtoprogramwithjava.com/hibernate-manytomany-unidirectional-bidirectional/

- Product is owning side as a product can exist without a routine but routines cannot exist without a product

User & Routine: https://www.baeldung.com/hibernate-one-to-many
> it's a good practice to mark many-to-one side as the owning side.
- Routine is owning side since many routines belong to one user

**Comment by egg-bread** [comment]
Ok so when a routine needs to be created (RoutinesService), I call setUser() and setProducts() on the routine. In the db, the user_fk field is populated correctly for routines table:
Screen Shot 2020-06-21 at 4 58 21 PM

When I retrieve the user's routines, I get the routines back but the products of the routine is empty.
Screen Shot 2020-06-21 at 4 57 59 PM

When I retrieve a product belonging to a routine, routines field is empty:
Screen Shot 2020-06-21 at 4 59 54 PM

There is also a sql error that happens with the following code in RoutinesService:
> Cannot add or update a child row: a foreign key constraint fails (`cellery_db`.`routine_products`, CONSTRAINT `fk_routines_routine_products` FOREIGN KEY (`id_routine`) REFERENCES `routines` (`id`) ON DELETE CASCADE ON UPDATE CASCADE)

Screen Shot 2020-06-21 at 5 01 48 PM

:( Something is wrong with how I am trying to update the relationship between Routines and Products.

**Comment by egg-bread** [comment]
Update: unidirectional relationship between Routine and Product (routine is child, product is parent)

**Comment by egg-bread** [comment]
Ok! Fixed the previous mapping issue between Routine & Product. I tested the new endpoints locally, they behave as expected but you can test it too to see how it worccs 😗

Also...I came to realize something...won't we also need a ManyToMany relationship between users and products? Because how else would we load all the products a user has 😳

File: src/main/java/com/cellery/api/backend/ui/controllers/ProductController.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        this.modelMapper = mapper;
+    }
+
+    @GetMapping
+    public String test() {
+        return "You have poked the /products endpoint";
+    }
+
+    @GetMapping(path="/get")
+    public ResponseEntity<ProductRespModel> getProduct(@Valid @RequestBody ProductRequestModel productReq) {
```
Comment (JckXia) [inline src/main/java/com/cellery/api/backend/ui/controllers/ProductController.java:43]: For the mappings, is it possible to have api routes such as
POST /products/users 
GET /products/users 
GET /products/users/{id}.
@GetMapping and such should be able to differentiate requests coming in even if they share the same route

File: src/main/java/com/cellery/api/backend/ui/controllers/RoutineController.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        return userDto;
+    }
+
+    @GetMapping
+    public String test() {
+        return "This is /routines endpoint";
+    }
+    // TODO: delete routine (all products will detach from routine), edit routine
+
+    @GetMapping(path="/get")
```
Comment (JckXia) [inline src/main/java/com/cellery/api/backend/ui/controllers/RoutineController.java:65]: Here as well

**Comment by egg-bread** [comment]
I THINK IT'S ALL DONE 😗 

I didn't try testing with multiple users yet

**Pull request merged**